### PR TITLE
fh apply: support requesting restricted tokens

### PR DESCRIPTION
**Pull request by cole-h**

### CODE REVIEW

File: src/cli/cmd/apply/mod.rs [hunk 1]
```diff
@@ -24,11 +25,20 @@ pub(crate) struct ApplySubcommand {
     #[clap(subcommand)]
     system: System,
 
+    #[clap(long)]
+    profile_path: Option<PathBuf>,
+
+    #[clap(long, default_value_t = false)]
+    request_restricted_token: bool,
```
Comment (grahamc) [inline src/cli/cmd/apply/mod.rs:32]:
```suggestion
    #[clap(long, default_value_t = true)]
    use_scoped_token: bool,
```

File: src/cli/cmd/apply/mod.rs [hunk 2]
```diff
@@ -24,11 +25,20 @@ pub(crate) struct ApplySubcommand {
     #[clap(subcommand)]
     system: System,
 
+    #[clap(long)]
+    profile_path: Option<PathBuf>,
```
Comment (grahamc) [inline src/cli/cmd/apply/mod.rs:29]: -1, we could do this as part of a `apply profile` subcommand, though.

**Comment by grahamc** [comment]
I was blocked on testing this end to end until resolve tokens can auth with `dnixd login token`, but testing individual pieces of this flow seems good. Can we fix the resolve tokens then re-up review of this?

File: src/cli/cmd/apply/mod.rs [hunk 3]
```diff
@@ -77,16 +85,94 @@ impl CommandExecute for ApplySubcommand {
...
+                return Err(color_eyre::eyre::eyre!(
+                    "FlakeHub did not return a restricted token!"
+                ));
+            }
+            Some(_) if !self.use_scoped_token => {
+                tracing::warn!(
+                    "Received a restricted token from FlakeHub, but we didn't request one! Ignoring."
+                );
+            }
+            _ => (),
```
Comment (lucperkins) [inline src/cli/cmd/apply/mod.rs:171]: Could you add a little note here saying what this catch-all case represents?

**Review by lucperkins — CHANGES_REQUESTED** [review]

**Review by lucperkins — APPROVED** [review]

**Review by colemickens — APPROVED** [review]

**Pull request merged**